sql-plugin/src/main/scala/org/apache/spark/sql/rapids/stringFunctions.scala

```diff
@@ -487,6 +487,88 @@ case class GpuInitCap(child: Expression) extends GpuUnaryExpression with Implici
     }
 }
 
+case class GpuStringRepeat(input: Expression, repeatTimes: Expression)
+    extends GpuBinaryExpression with ImplicitCastInputTypes with NullIntolerant {
+  override def left: Expression = input
+  override def right: Expression = repeatTimes
+  override def dataType: DataType = input.dataType
+  override def inputTypes: Seq[AbstractDataType] = Seq(StringType, IntegerType)
+
+  def doColumnar(input: GpuScalar, repeatTimes: GpuColumnVector): ColumnVector = {
+    assert(input.dataType == StringType)
+
+    withResource(GpuColumnVector.from(input, repeatTimes.getRowCount.asInstanceOf[Int],
+                 input.dataType)) {
+      replicatedInput => doColumnar(replicatedInput, repeatTimes)
+    }
+  }
+
+  def doColumnar(input: GpuColumnVector, repeatTimes: GpuColumnVector): ColumnVector = {
+    val repeatTimesCV = repeatTimes.getBase
+
+    // Compute the output size to check for overflow.
+    withResource(input.getBase.repeatStringsSizes(repeatTimesCV)) { outputSizes =>
+      if (outputSizes.getTotalSize > Int.MaxValue.asInstanceOf[Long]) {
+        throw new RuntimeException("Output strings have total size exceed maximum allowed size")
+      }
+
+      // Finally repeat the strings using the pre-computed strings' sizes.
+      input.getBase.repeatStrings(repeatTimesCV, outputSizes.getStringSizes)
+    }
+  }
+
+  def doColumnar(input: GpuColumnVector, repeatTimes: GpuScalar): ColumnVector = {
+    assert(repeatTimes.dataType == IntegerType)
+
+    if (!repeatTimes.isValid) {
+      // If the input scala repeatTimes is invalid, the results should be all nulls.
+      // This is equivalent to having repeatTimes == 0.
+      input.getBase.repeatStrings(0)
+    } else {
+      repeatTimes.getValue match {
```

Comment:
Like is done for the other types here, I'd just add an `assert(repeatTimes.dataType == IntegerType)` and call `repeatTimes.getBase.getInt`.  Spark will already be checking that the input is IntegerType while creating the logical plan.